lib/matplotlib/colorbar.py

```diff
@@ -1090,8 +1090,13 @@ def _process_values(self):
             # If we still aren't scaled after autoscaling, use 0, 1 as default
             self.norm.vmin = 0
             self.norm.vmax = 1
-        self.norm.vmin, self.norm.vmax = mtransforms.nonsingular(
-            self.norm.vmin, self.norm.vmax, expander=0.1)
+
+        # Only expand vmax if needed, to match Normalize's behavior of mapping
+        # everything to 0 if the norm is singular.
+        vmin, vmax = sorted([self.norm.vmin, self.norm.vmax])
+        self.norm.vmin = vmin
+        _, self.norm.vmax = mtransforms.nonsingular(vmin, vmax, expander=0.1)
```

Comment:
This forces `self.norm.vmin <= self.norm.vmax` now I think. Would a user care if they had defined their norm reversed and we are now swapping this for them?

Reply:
The previous version (just calling nonsingular) already performed that ordering; I'm not changing that behavior (which I agree we can question), just doing it explicitly first to make it easier to just expand on the top side.